dScripts/WhFans.cpp

```diff
@@ -0,0 +1,66 @@
+#include "WhFans.h"
+
+#include "RenderComponent.h"
+
+void WhFans::OnStartup(Entity* self) {
+	self->SetVar<bool>(u"alive", true);
+	self->SetVar<bool>(u"on", false);
+
+	ToggleFX(self, false);
+}
+
+void WhFans::ToggleFX(Entity* self, bool hit) {
+	std::string fanGroup = "";
+	try {
+		fanGroup = self->GetGroups()[0];
+	} catch(...) {
+		fanGroup = "";
+	}
```

Comment:
Do not use catch (...) unless absolutely necessary. Do something like
```c++
std::string fanGroup;
const auto& groups = self->GetGroups();
if (!groups.empty()) {
    fanGroup = groups[0];
} else {
    fanGroup = "";
}
```

Reply:
Noted, still experimenting with the fans. Currently in a state where it causes the the world to crash on load and was trying to get it to not